src/shortest_path/all_pairs_dijkstra.rs

```diff
@@ -81,23 +81,24 @@ pub fn all_pairs_dijkstra_path_lengths<Ty: EdgeType + Sync>(
         .into_par_iter()
         .map(|x| {
             let out_map = PathLengthMapping {
-                path_lengths: dijkstra(
+                path_lengths: (dijkstra(
                     graph,
                     x,
                     None,
                     |e| edge_cost(e.id()),
                     None,
                 )
-                .unwrap()
-                .iter()
-                .filter_map(|(index, cost)| {
-                    if *index == x {
-                        None
-                    } else {
-                        Some((index.index(), *cost))
-                    }
-                })
-                .collect(),
+                    as PyResult<DictMap<NodeIndex, f64>>)
```

Comment:
Type casting here doesn't look great (in terms of readability). IMO, type annotation is nicer:

```rust
let path_lengths: PyResult<DictMap<NodeIndex, f64>> =
     dijkstra(graph, x, None, |e| edge_cost(e.id()), None);

let out_map = PathLengthMapping {
     path_lengths: path_lengths
            .unwrap()
            .iter()
            ...
            .collect(),
};
```